src/components/Query.ts

```diff
@@ -0,0 +1,120 @@
+import * as vscode from "vscode";
+import rp = require("request-promise");
+import { InfluxDBConnection } from "./connections/Connection";
+import { Row, TableView } from "./TableView";
+import { Status } from "./connections/Status";
+import { INode } from "./connections/INode";
+
+export class Engine {
+  private outputChannel: vscode.OutputChannel;
+
+  public constructor(outputChannel: vscode.OutputChannel) {
+    this.outputChannel = outputChannel;
+  }
+
+  public async GetTreeChildren(
+    conn: InfluxDBConnection,
+    query: string,
+    msg: string,
+    newNodeFn: (
+      name: string,
+      outputChannel: vscode.OutputChannel,
+      iConn: InfluxDBConnection
+    ) => INode
+  ): Promise<INode[]> {
+    this.outputChannel.show();
+    this.outputChannel.appendLine(msg);
+    let result: Result = await APIRequest.Query(conn, query);
+    if (result.Err !== undefined) {
+      this.outputChannel.appendLine("Err: " + result.Err);
+    } else {
+      result.Rows = (result.Rows as Array<Row>).slice(1);
+      var nodes: Array<INode> = [];
+      for (let row of result.Rows as Array<Row>) {
+        nodes.push(newNodeFn(row.Field[0], this.outputChannel, conn));
+      }
+      return nodes;
+    }
+    return [];
+  }
+
+  public async TableView(iConn?: InfluxDBConnection, query?: string) {
+    const activeTextEditor = vscode.window.activeTextEditor;
```

Comment:
Just a style thing, but usually cleaner to just deconstruct with things like this